src/main/java/cpsc/dlsproject/visitors/ParseVisitor.java

```diff
@@ -0,0 +1,289 @@
+                    break;
+
+                case "VAR":
+                    statements.add((Statement) this.visit(new VarDeclaration()));
+                    break;
+
+                case "IF":
+                    statements.add((Conditional) this.visit(new Conditional()));
+                    break;
+
+                default:
+                    throw new Exception("Invalid statement in body of endpoint declaration.");
+            }
+            if (tokenizer.checkCurrent().equals(";")) {
+                tokenizer.getAndCheckNext(";");
+            }
+        }
+        return statements;
+    }
+
+    @Override
+    BaseAST visit(VarAccess varAccess) throws Exception {
+        varAccess.identifier = tokenizer.getNext();
+        return varAccess;
+    }
+
+    @Override
+    BaseAST visit(Conditional conditional) throws Exception {
+        // TODO: Add logic for expression.
+        tokenizer.getAndCheckNext("IF");
+        tokenizer.getAndCheckNext("\\(");
+        conditional.condition = parseExpression();
+        tokenizer.getAndCheckNext("\\)");
+        tokenizer.getAndCheckNext("\\{");
+        conditional.thenCase = parseStatements();
+        tokenizer.getAndCheckNext("\\}");
+        if (tokenizer.checkNext().equals("ELSE")) {
+            tokenizer.getAndCheckNext("ELSE");
+            tokenizer.getAndCheckNext("\\{");
+            conditional.elseCase = parseStatements();
+            tokenizer.getAndCheckNext("\\}");
+        }
+
+        return conditional;
+    }
+
+    @Override
+    BaseAST visit(Response response) throws Exception {
+        tokenizer.getAndCheckNext("SEND");
+        tokenizer.getAndCheckNext("\\{");
+        response.statusCode = Integer.parseInt(tokenizer.getNext());
+        tokenizer.getAndCheckNext(";");
+        response.message = tokenizer.getNext();
+        tokenizer.getAndCheckNext(";");
+        tokenizer.getAndCheckNext("\\}");
+        return response;
+    }
+
+    @Override
+    BaseAST visit(URLDeclaration url) throws Exception {
+        String urlString = tokenizer.getNext();
+        if (!utils.checkIfTokenIsString(urlString)) {
+            throw new Exception("url is not string");
+        }
+
+        url.url = utils.getStringLiteralContent(urlString);
+        return url;
+    }
+
+    @Override
+    BaseAST visit(VarDeclaration varDeclaration) throws Exception {
+        tokenizer.getAndCheckNext("VAR");
+        String name = tokenizer.getNext();
+        if (name.charAt(name.length() - 1) != ':') {
+            throw new Exception("variable declaration must have colon to separate identifier and the type.");
+        }
+        varDeclaration.name = name.substring(0, name.length() - 1);
+
+        String type = tokenizer.getNext();
+
+        switch (type) {
+            case "Number":
+                varDeclaration.type = Type.NUMBER;
+                varDeclaration.expression = new NumberValue(Double.parseDouble(tokenizer.getNext()));
```

Comment:
This can be a variable defined before. Eg:
VAR a : Number = 2;
VAR b : Number = a + 2;